### PR TITLE
feat(resource limits): add resource limit in the local cli

### PR DESCRIPTION
**Pull request by fengwa-aws**
**Issue #, if available:**

**Description of changes:**
Enable local cli users to configure resource limits in the local deployment
**Why is this change necessary:**

**How was this change tested:**
unit test

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java [hunk 1]
```diff
@@ -60,13 +64,12 @@ public DeploymentCommand(NucleusAdapterIpc nucleusAdapterIpc) {
         if (configUpdate != null && !configUpdate.isEmpty()) {
             try {
                 // Try to read JSON from a file if it is a path and the file exists
-                try {
-                    Optional<Path> filePath = deTilde(configUpdate);
-                    if (filePath.isPresent() && Files.exists(filePath.get())) {
-                        configurationUpdate = mapper.readValue(filePath.get().toFile(), Map.class);
-                    }
-                } catch (InvalidPathException ignored) {
```
Comment (MikeDombo) [inline client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:68]: don't remove the catch

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java [hunk 2]
```diff
@@ -80,12 +83,29 @@ public DeploymentCommand(NucleusAdapterIpc nucleusAdapterIpc) {
             }
         }
 
+        MapType mapType = mapper.getTypeFactory().constructMapType(HashMap.class, String.class,
+                SystemResourceLimits.class);
+        Map<String, SystemResourceLimits> systemResourceLimits = new HashMap<>();
+        try {
+            Optional<Path> filePath = deTilde(systemLimits);
```
Comment (MikeDombo) [inline client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:90]: so you're only supporting a file input, what about command line?

Comment (fengwa-aws) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:90]: Do we need to support a command line json iput? Doesn't seems user-friendly.

Comment (MikeDombo) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:90]: Why isn't that user friendly to allow more than one option? Command line is probably necessary for our tests anyway. 

We support both file and cmd for the configuration input.

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java (hunk 1 above)
Comment (fengwa-aws) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:68]: which line is throwing this exception?

Comment (MikeDombo) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:68]: `deTilde`

**Review by MikeDombo — APPROVED** [review]

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java [hunk 3]
```diff
@@ -80,12 +86,40 @@ public DeploymentCommand(NucleusAdapterIpc nucleusAdapterIpc) {
             }
         }
 
+        MapType mapType = mapper.getTypeFactory().constructMapType(HashMap.class, String.class,
+                SystemResourceLimits.class);
+        Map<String, SystemResourceLimits> systemResourceLimits = new HashMap<>();
+        if (systemLimits != null && !systemLimits.isEmpty()) {
```
Comment (leaf94) [inline client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:92]: nit - `Utils.isEmpty`?

Comment (fengwa-aws) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:92]: The utils library is not brought in the cli. I don't feel strongly it's necessary.

**Review by leaf94 — APPROVED** [review]

**Pull request merged**